odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+
+    The source moves on a circle with radius ``r``, and the detector
+    reference point is opposite to the source on a circle with radius ``R``
+    and aligned tangential to the circle.
+
+    The motion parameter is the (1d) rotation angle parametrizing source and
+    detector positions.
+    """
+
+    def __init__(self, angle_intvl, dparams, src_rad, det_rad, agrid=None,
+                 dgrid=None, angle_offset=0, axis=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        angle_intvl : `Interval` or 1-dim. `IntervalProd`
+            The motion parameters
+        dparams : `Rectangle` or 2-dim. `IntervalProd`
+            The detector parameters
+        src_rad : `float`
+            Radius of the source circle, must be positive
+        det_rad : `float`
+            Radius of the detector circle, must be positive
+        agrid : 1-dim. `TensorGrid`, optional
+            A sampling grid for `angle_intvl`. Default: `None`
+        dgrid : 2-dim. `TensorGrid`, optional
+            A sampling grid for ``dparams``. Default: `None`
+        angle_offset : `float`, optional
+            Offset to the rotation angle in the azimuthal plane. Does not
+            imply an offset in z-direction. Default: 0
+        axis : `int` or 3-element array, optional
+            Defines the rotation axis via a 3-element vector or a single
+            integer referring to a standard axis. Default: `None`
+        """
+
+        super().__init__(angle_intvl, dparams, src_rad, det_rad, agrid,
+                         dgrid, angle_offset, axis)
+
+    def det_refpoint(self, angle):
+        """The detector reference point function.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+            The reference point on the circle with radius ``R`` at a given
+            rotation angle :math:`\\phi`, defined as :math:`R(-\\sin\\phi,
+            \\cos\\phi, 0)`
+        """
+        if angle not in self.motion_params:
+            raise ValueError('angle {} not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+
+        angle += self.angle_offset
+
+        # ASTRA 'cone_vec' convention
+        return self.det_radius * np.array([-sin(angle), cos(angle), 0])
+
+    def src_position(self, angle):
+        """The source position function.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+
+            The source position at ``z`` on the circle with radius ``r`` at
+            a given rotation angle :math:`\\phi`, defined as :math:`(r
```

Comment:
We did in the first place but ASTRA doesn't. I'll changed it to get the helical geometry running and make the cone beam geometries at least consistent. But sure, this has to be rectified and made consistent everywhere.